zhaquirks/niko/switch.py

```diff
@@ -0,0 +1,313 @@
+"""Niko switches (only Niko Double connectable switch,10A (552-721X2) at this time)"""
+
+from zigpy import types as t
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+)
+from zigpy.zcl.clusters.homeautomation import Diagnostic
+
+from zhaquirks import EventableCluster
+from zhaquirks.const import (
+    ARGS,
+    BUTTON_1,
+    BUTTON_2,
+    BUTTON_3,
+    BUTTON_4,
+    COMMAND_CLICK,
+    COMMAND_HOLD,
+    COMMAND_RELEASE,
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.niko import NIKO
+
+
+class NikoConfigCluster(CustomCluster):
+    """manufacturer specific cluster related to device settings"""
+
+    class ButtonOperationMode(t.enum8):
+        # WARN: while one can technically write 0x00 to the operationMode attribute
+        #       this seems to brick the device and it will need to be rejoined
+        Decoupled = 0x01
+        ControlRelay = 0x02
+
+    class LedState(t.uint8_t):
+        Off = 0x00
+        On = 0xFF
+
+    class LedOperationMode(t.enum8):
+        Decoupled = 0x00
+        ControlledByRelay = 0x01
+        # Notable behaviour:
+        # If LedOperationMode == ControlledByRelay, ButtonOperationMode == ControlRelay and LedState == On, then LED reflects status of corresponding relay
+        # If ButtonOperationMode == Decoupled, then neither buttons nor relay affect the LEDs (i.e. LedOperationMode acts as if ==Decoupled)
+        # If LedOperationMode == Decoupled and/or ButtonOperationMode == Decoupled, then LED can nevertheless by controlled by writing LedState ('always on' or 'always off')
+
+    cluster_id = 0xFC00
+    ep_attribute = "niko_config_cluster"
+    attributes = {
+        # If this would ever be expanded to outlets:
+        # WARNING: 0x0000 has different datatypes!
+        # enum8 (switch) vs. bitmap8 (outlet)
+        # unknown usage/function on outlet
+        0x0000: ("button_operation_mode", ButtonOperationMode, True),
```

Comment:
Ideally, we should switch all attributes to use the new-style zigpy `AttributeDefs`, similar to this:
https://github.com/zigpy/zha-device-handlers/blob/ab0d248defda933019c98ddab451b5969fe99c16/zhaquirks/legrand/cable_outlet.py#L42-L59

Reply:
@TheJulianJES 
Do you have an other example of this?
I tried changing the attributes to
```
class AttributeDefs(BaseAttributeDefs):
    button_operation_mode = ZCLAttributeDef(
        id=0x0000,
        type=t.enum8,
        is_manufacturer_specific=True,
    )
    led_state = ZCLAttributeDef(
        id=0x0100,
        type=t.uint8_t,
        is_manufacturer_specific=True,
    )
    led_operation_mode = ZCLAttributeDef(
        id=0x0104,
        type=t.enum8,
        is_manufacturer_specific=True,
    )
```
But then I get `enum8.undefined_0x01` when reading the attribute in homeassistant ui.
![image](https://github.com/zigpy/zha-device-handlers/assets/12485043/b05f27a4-9b9b-4689-8877-bbf066bde718)

Changing type to `ButtonOperationMode` does not seem to work either.

Reply:
`type = ButtonOperationMode` (and so on) would be correct and should work.
There are more examples in the zigpy project in the `zcl` folder where all ZCL clusters are defined.

Reply:
@TheJulianJES When defining the `ButtonOperationMode` class outside the `NikoConfigCluster` class and adding `ButtonOperationMode: Final = ButtonOperationMode` in the `NikoConfigCluster`, it works.